projet/projects/models.py

```diff
@@ -76,3 +123,73 @@ def save(self, *args, **kwargs):
                 raise ValueError("Due date cannot be in the past.")
 
         super().save(*args, **kwargs)
+
+
+class Budget(models.Model):
+    FLEXIBILITY_CHOICES = [("F", "Flexible"), ("T", "Tight")]
+    amount = MoneyField(
+        max_digits=14, decimal_places=2, default_currency="USD", null=False, blank=False
+    )
+    project = models.ForeignKey(
+        Project,
+        blank=False,
+        null=False,
+        on_delete=models.CASCADE,
+        related_name="budget",
+    )
+    date_created = models.DateTimeField(auto_now_add=True)
+    budget_type = models.CharField(
+        choices=FLEXIBILITY_CHOICES, max_length=10, default="F"
+    )
+
+    def get_project(self):
+        return Project.objects.get(id=self.project.id)
+
+    def __str__(self):
+        return f"{self.amount_currency}{self.amount}"
+
+
+class BudgetHistory(models.Model):
+    budget = models.ForeignKey(
+        Budget,
+        on_delete=models.CASCADE,
+        blank=False,
+        null=False,
+        related_name="history",
+    )
+    from_amount = MoneyField(
+        max_digits=14, decimal_places=2, default_currency="USD", null=True, blank=True
+    )
+    to_amount = MoneyField(
+        max_digits=14, decimal_places=2, default_currency="USD", null=True, blank=True
+    )
+    date_updated = models.DateTimeField(default=timezone.now())
+
+    def determine(self):
+        if self.to_amount < self.from_amount:
+            return "Decrease"
+        elif self.to_amount > self.from_amount:
```

Comment:
This is the function that determines if the changes are increases or decreases. It is used by the API.